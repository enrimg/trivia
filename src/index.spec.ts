// eslint-disable-next-line import/no-extraneous-dependencies
import runGoldenMaster from "jest-golden-master";
import { Game } from ".";

test("All correct scenario", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(6);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(4);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(3);
    game.wasCorrectlyAnswered();
  });
});
test("Wrong answer scenario", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(6);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(4);
    game.wrongAnswer();
  });
});

test("Wrong answer and even roll with correct answer scenario", async () => {
  runGoldenMaster(async () => {
    const game = new Game();
    game.add("Mathieu");
    game.add("Thomas");
    game.roll(6);
    game.wasCorrectlyAnswered();
    game.roll(1);
    game.wasCorrectlyAnswered();
    game.roll(4);
    game.wrongAnswer();
    game.roll(4);
    game.wasCorrectlyAnswered();
    game.roll(4);
  });
});
