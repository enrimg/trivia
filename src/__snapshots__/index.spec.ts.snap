// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All correct scenario 1`] = `
Array [
  "Mathieu is the current player",
  "player Mathieu placed in 6",
  "Sports Question 0",
  "this.currentPlayer 0 answered correctly",
  "player Mathieu wins no",
  "next player is Thomas",
  "Thomas is the current player",
  "player Thomas placed in 1",
  "Science Question 0",
  "this.currentPlayer 1 answered correctly",
  "player Thomas wins no",
  "next player is Mathieu",
  "Mathieu is the current player",
  "player Mathieu placed in 10",
  "Sports Question 1",
  "this.currentPlayer 0 answered correctly",
  "player Mathieu wins no",
  "next player is Thomas",
  "Thomas is the current player",
  "player Thomas placed in 2",
  "Sports Question 2",
  "this.currentPlayer 1 answered correctly",
  "player Thomas wins no",
  "next player is Mathieu",
  "Mathieu is the current player",
  "player Mathieu placed in 1",
  "Science Question 1",
  "this.currentPlayer 0 answered correctly",
  "player Mathieu wins no",
  "next player is Thomas",
]
`;

exports[`Wrong answer and even roll with correct answer scenario 1`] = `
Array [
  "Mathieu is the current player",
  "player Mathieu placed in 6",
  "Sports Question 0",
  "this.currentPlayer 0 answered correctly",
  "player Mathieu wins no",
  "next player is Thomas",
  "Thomas is the current player",
  "player Thomas placed in 1",
  "Science Question 0",
  "this.currentPlayer 1 answered correctly",
  "player Thomas wins no",
  "next player is Mathieu",
  "Mathieu is the current player",
  "player Mathieu placed in 10",
  "Sports Question 1",
  "Question was incorrectly answered, player Mathieu gets sent to the penalty box",
  "next player is Thomas",
  "Thomas is the current player",
  "player Thomas placed in 5",
  "Science Question 1",
  "this.currentPlayer 1 answered correctly",
  "player Thomas wins no",
  "next player is Mathieu",
  "Mathieu is the current player",
]
`;

exports[`Wrong answer scenario 1`] = `
Array [
  "Mathieu is the current player",
  "player Mathieu placed in 6",
  "Sports Question 0",
  "this.currentPlayer 0 answered correctly",
  "player Mathieu wins no",
  "next player is Thomas",
  "Thomas is the current player",
  "player Thomas placed in 1",
  "Science Question 0",
  "this.currentPlayer 1 answered correctly",
  "player Thomas wins no",
  "next player is Mathieu",
  "Mathieu is the current player",
  "player Mathieu placed in 10",
  "Sports Question 1",
  "Question was incorrectly answered, player Mathieu gets sent to the penalty box",
  "next player is Thomas",
]
`;
